src/instructlab/utils.py

```diff
@@ -556,15 +566,89 @@ def split_hostport(hostport: str) -> tuple[str, int]:
     return hostname, port
 
 
-def convert_messages_to_legacy_dataset(
+def is_pretraining_dataset(ds: List[MessageSample]) -> bool:
+    """
+    Determines whether or not the given dataset is the phase007 pretraining
+    dataset.
+    """
+    if not ds:
+        return False
+    sample = ds[0]
+    return any(m["role"] == "pretraining" for m in sample["messages"])
+
+
+def get_user_assistant_from_pretraining(pretraining: str) -> Tuple[str, str]:
+    """
+    Given a pretraining message sample which contains the tokens '<|user|>' and '<|assistant|>',
+    return the strings corresponding to the user & assistant inputs.
+
+    This function assumes that a given message begins with the '<|user|>' token
+    and contains '<|assistant|>' somewhere later.
+
+    Raises ValueError if neither the '<|user|>' or '<|assistant|>' tokens exist
+    within the given sample.
+    """
+    if SDGTokens.ASSISTANT not in pretraining:
```

Comment:
nit:

```
for token in (ASSISTANT, USER):
    if token not in pretraining:
         raise ...
```

Reply:
That's a bit cleaner, I'll do this instead.